nkms_eth/escrow.py

```diff
@@ -1,78 +1,178 @@
 import random
-from typing import List
-from nkms_eth import blockchain
-from nkms_eth import token
-
-ESCROW_NAME = 'MinersEscrow'
-PREMINE = int(1e9) * token.M
-REWARD = token.SATURATION - PREMINE
-HOURS_PER_PERIOD = 1  # 24
-MIN_RELEASE_PERIODS = 1  # 30
-MAX_AWARDED_PERIODS = 365
-MIN_ALLOWED_LOCKED = 10 ** 6
-MAX_ALLOWED_LOCKED = 10 ** 7 * token.M
-MINING_COEFF = [
-    HOURS_PER_PERIOD,
-    2 * 10 ** 7,
-    MAX_AWARDED_PERIODS,
-    MAX_AWARDED_PERIODS,
-    MIN_RELEASE_PERIODS,
-    MIN_ALLOWED_LOCKED,
-    MAX_ALLOWED_LOCKED
-]
-NULL_ADDR = '0x' + '0' * 40
-
-
-def create():
-    """
-    Creates an escrow which manages mining
-    """
-    chain = blockchain.chain()
-    creator = chain.web3.eth.accounts[0]  # TODO: make it possible to override
-    tok = token.get()
-    escrow, tx = chain.provider.get_or_deploy_contract(
-        ESCROW_NAME, deploy_args=[token.get().address] + MINING_COEFF,
-        deploy_transaction={'from': creator})
-    chain.wait.for_receipt(tx, timeout=blockchain.TIMEOUT)
-    tx = tok.transact({'from': creator}).transfer(escrow.address, REWARD)
-    chain.wait.for_receipt(tx, timeout=blockchain.TIMEOUT)
-    tx = escrow.transact({'from': creator}).initialize()
-    chain.wait.for_receipt(tx, timeout=blockchain.TIMEOUT)
-    return escrow
-
-
-def get():
-    """
-    Returns an escrow object
-    """
-    return token.get(ESCROW_NAME)
+from typing import List, Tuple, Set, Generator
+
+from populus.contracts.contract import PopulusContract
+
+from nkms_eth.token import NuCypherKMSToken
+from .blockchain import Blockchain
 
+addr = str
 
-def sample(n: int = 10)-> List[str]:
+
+class Escrow:
     """
-    Select n random staking Ursulas, according to their stake distribution
-    The returned addresses are shuffled, so one can request more than needed and
-    throw away those which do not respond
+    In order to become a participant of the network,
+    a miner locks tokens by depositing to the Escrow contract
```

Comment:
OK, so here I read that this is the "Escrow contract" - as if maybe I'm looking at a quasi-inheritance pattern, but without any literal class derivation.  But then...

Reply:
More of a wrapper...

Reply:
Updated.